src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/SvDiscoverFromLocalAssemblyContigAlignmentsSpark.java

```diff
@@ -272,44 +281,197 @@ public static AssemblyContigsClassifiedByAlignmentSignatures preprocess(final Sv
      * {@link AssemblyContigWithFineTunedAlignments.AlignmentSignatureBasicType#UNKNOWN}
      * currently DO NOT generate any VCF yet.
      */
-    public static void dispatchJobs(final JavaSparkContext ctx,
-                                    final AssemblyContigsClassifiedByAlignmentSignatures contigsByPossibleRawTypes,
-                                    final SvDiscoveryInputMetaData svDiscoveryInputMetaData,
-                                    final JavaRDD<GATKRead> assemblyRawAlignments,
-                                    final boolean writeSAMFiles) {
+    public static List<VariantContext> dispatchJobs(final JavaSparkContext ctx,
+                                                    final AssemblyContigsClassifiedByAlignmentSignatures contigsByPossibleRawTypes,
+                                                    final SvDiscoveryInputMetaData svDiscoveryInputMetaData,
+                                                    final JavaRDD<GATKRead> assemblyRawAlignments,
+                                                    final boolean writeSAMFiles) {
 
         final String outputPrefixWithSampleName = svDiscoveryInputMetaData.getOutputPath();
 
-        // TODO: 1/10/18 bring back read annotation, see ticket 4228
+        final List<VariantContext> simpleChimeraVariants =
+                extractSimpleVariants(contigsByPossibleRawTypes.simple, svDiscoveryInputMetaData, outputPrefixWithSampleName);
+        contigsByPossibleRawTypes.simple.unpersist(false);
```

Comment:
I think you just moved around this code for this PR, but looking at this I'm not sure I like this method having the side effect of calling unpersist on the internal RDDs of `contigsByPossibleRawTypes`. Just to avoid side effects I'd prefer to pull out the `unpersist` calls to the calling method. Actually, since you've got them bundled in one object, why not make an `unpersist` method on `AssemblyContigsClassifiedByAlignmentSignatures` that unpersists all of the underlying RDDs, and then it's just a one line invocation from the callers of this method.

Reply:
Thanks! That's much better.